homework/index.js

```diff
@@ -0,0 +1,144 @@
+            renderContributions(contributors);
+        } catch (err) {
+            renderError(err);
+        }
+
+    }
+
+    function fetchJSON(url) {
+
+        return new Promise((resolve, reject) => {
+            const XHR = new XMLHttpRequest();
+            XHR.open('GET', url, true);
+            XHR.responseType = 'json';
+            XHR.onload = () => {
+                if (XHR.status < 400) {
+                    // resolve the promise if the request is OK
+                    resolve(XHR.response);
+
+                } else {
+                    // if bad response reject the promise with error object
+                    reject(new Error(`Network error: ${XHR.status} - ${XHR.statusText}`));
+
+                }
+            };
+            //if XHR not loaded reject the promise
+            XHR.onerror = () => reject(new Error('Network request failed'));
+            XHR.send();
+        });
+
+    }
+
+    async function renderReposList(reposObj) {
+
+        const header = document.querySelector('header');
+        reposObj.sort((a, b) => { return a.name.localeCompare(b.name); });
+        //create list and append it to header
+        const selectList = createAndAppend('select', header, { id: 'selectList' });
+
+        for (const rep in reposObj) {
+            // add an option for each repo with value that is his index at reposObject 
+            createAndAppend('option', selectList, { value: rep, text: reposObj[rep].name });
+
+        }
+        //add a listener when select a new option
+        selectList.onchange = async () => {
+
+            try {
+                const contributors = await fetchJSON(reposObj[selectList.value].contributors_url);
+                renderInfo(reposObj[selectList.value]);
+                renderContributions(contributors);
+            } catch (err) {
+                renderError(err);
+            }
+
+        };
+        return reposObj[selectList.value];
+
+    }
+
+    function renderInfo(rep) {
+
+        mainContainer.innerText = '';
+        const infoSection = createAndAppend('section', mainContainer, { id: 'info-section' });
+        const table = createAndAppend('table', infoSection); // create info table 
+        const infoObj = { 'Repository': '', 'Description': rep.description, 'Forks': rep.forks, 'Update': rep.updated_at.substring(0, 10) }; //create 4 rows * 2 cell
+
+        for (const atr in infoObj) {
+            const row = table.insertRow();
+            row.insertCell(0).innerText = atr;
+            row.insertCell(1).innerText = infoObj[atr];
+        }
+
+        createAndAppend('a', document.querySelector('#info-section tr:nth-child(1) td:nth-child(2)'),
+            { href: rep.html_url, target: '_blank', text: rep.name });
+
+    }
+
+    function renderError(error) {
+        mainContainer.innerText = '';
+        createAndAppend('div', mainContainer, { text: error.message, class: 'error' });
+    }
+
+    function renderContributions(contributions) {
```

Comment:
Your function `renderContributions` might perhaps better be called `renderContributors`, to avoid confusion with the `contributions` property, which is just a number. Similarly, the parameter name could be `contributors`. Note that this parameter receives an array, so the line `if (contributions)` serves no purpose. An empty array is also `truthy`, so the condition is always `true`. So you can leave it out as the `for` will be executed zero times if the array is empty.

Reply:
Actually here I'm trying to cover the case when the repo is empty as [hyfer-infra](https://github.com/HackYourFuture/hyfer-infra) so the contributors_url return `null`.
I corrected the wrong named function and parameter.

Reply:
Actually here I'm trying to cover the case when the repo is empty as hyfer-infra so the contributors_url return null.
I corrected the wrong named function and parameter.

Reply:
You are completely right. My own app shows this for that repo:

![hyfer-infra](https://user-images.githubusercontent.com/2788771/42102510-6573f582-7bc6-11e8-8b43-ffb038ca5f75.png)